src/opening-period/page/EditOpeningPeriod.tsx

```diff
@@ -0,0 +1,301 @@
+      <Datepicker
+        id={name}
+        dataTest={name}
+        labelText={label}
+        onChange={onChange}
+        value={new Date(value)}
+        registerFn={register}
+        required
+      />
+    )}
+  />
+);
+
+export default function EditOpeningPeriodPage({
+  resourceId,
+  datePeriodId,
+}: {
+  resourceId: string;
+  datePeriodId: string;
+}): JSX.Element {
+  const id = parseInt(datePeriodId, 10);
+  const selectedLanguage: Language = 'fi';
+  const [resource, setResource] = useState<Resource>();
+  const [hasLoadingResourceError, setLoadingResourceError] = useState<
+    Error | undefined
+  >(undefined);
+  const [hasDatePeriodLoadingError, setDatePeriodLoadingError] = useState<
+    Error | undefined
+  >(undefined);
+  const { reset, register, handleSubmit, control, errors } = useForm<
+    FormData
+  >();
+  const history = useHistory();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [submitStatus, setSubmitStatus] = useState<SubmitStatus>('init');
+  const resetForm = useCallback(
+    (datePeriod: DatePeriod): void =>
+      reset({
+        openingPeriodTitle: datePeriod.name[selectedLanguage] || '',
+        openingPeriodOptionalDescription:
+          datePeriod.description[selectedLanguage] || '',
+        openingPeriodBeginDate: datePeriod.start_date || undefined,
+        openingPeriodEndDate: datePeriod.end_date || undefined,
+      }),
+    [reset]
+  );
+
+  const onSubmit = async (data: FormData): Promise<void> => {
+    try {
+      if (!resource || !data) {
+        throw new Error('This should never happen, but typescript');
+      }
+
+      setIsSaving(true);
+      const updatedPeriod = await api.putDatePeriod({
+        resource: resource.id,
+        id,
+        name: {
+          fi: data.openingPeriodTitle,
+          sv: null,
+          en: null,
+        },
+        description: {
+          fi: data.openingPeriodOptionalDescription,
+          sv: null,
+          en: null,
+        },
+        start_date: transformToApiFormat(data.openingPeriodBeginDate),
+        end_date: transformToApiFormat(data.openingPeriodEndDate),
+        resource_state: ResourceState.OPEN,
+        override: false,
+        time_span_groups: [],
+      });
+      resetForm(updatedPeriod);
+      setIsSaving(false);
+      setSubmitStatus('succeeded');
+    } catch (err) {
+      setIsSaving(false);
+      setSubmitStatus('error');
+      throw err;
+    }
+  };
+
+  useEffect((): void => {
+    // UseEffect's callbacks are synchronous to prevent a race condition.
+    // We can not use an async function as an useEffect's callback because it would return Promise<void>
+    api
+      .getResource(resourceId)
+      .then((r: Resource) => {
+        setResource(r);
+      })
+      .catch((e: Error) => {
+        setLoadingResourceError(e);
+        setIsLoading(false);
+      });
+  }, [resourceId]);
+
+  useEffect((): void => {
```

Comment:
Is it best practice to have multiple useEffect calls, or to try to do everything inside one?

Reply:
This is fixed.

Reply:
Good job with this!